### ISSUE TITLE
Login in multiple envs is not possible

### ISSUE THREAD

**Issue by rbarilani** [zalando-incubator/solution-center-login#49]
I can't login in the same browser session in multiple envs with different users because SC_TOKEN cookie is global and doesn't distinguish between envs.

Scenario to reproduce the bug:
Login in sc-stage then go to sc-integration.
You are logged out also from the stage env because the software takes SC_TOKEN try to validate that against integration token endpoint and since is not valid (it's the stage token), it clears the cookie (something that IHMO we should avoid in this case, see #32).

My quick and dirty proposal...save cookies with different names for different envs:
   SC_TOKEN : for production
   SC_TOKEN_<ENV_NAME> : for the rest

@dami-gg @gregdeane @cjhowald ... WDYT?

**Comment by TomTomaso** [comment]
i will need this, maybe also soon for my documentation app, to login multiple enviroments at the same time.

**Comment by cjhowald** [comment]
@rbarilani @TomTomaso To be clear, is this what you had in mind?

prod: SC_TOKEN (same)
stage: SC_TOKEN_STAGE
integration: SC_TOKEN_INTEGRATION
dev: SC_TOKEN_DEV

**Comment by rbarilani** [comment]
@cjhowald Yes.  I don't totally like it but at the moment I don't have anything else in mind.

**Comment by gregdeane** [comment]
@rbarilani @cjhowald I'm thinking something like this. This will mean that the dev cookie will be `SC_TOKEN_DEVELOPMENT`:

```javascript
function authenticationFactory($q, $localStorage, $cookies, environmentsService, $window, $injector, jwtHelper, $location, $timeout) {
  'use strict';

  var self = this;
  var envName = self.getEnvironment().NAME;
  var cookieEnv = (envName !== 'PRODUCTION' && '_'.concat(envName)) || '';

  var TOKEN_COOKIE_KEY = "SC_TOKEN" + cookieEnv;
  var BRAND_COOKIE_KEY = "SC_BRAND";
```

The above is abbreviated code, of course.

**Comment by rbarilani** [comment]
@gregdeane As long we use the public api we can name the cookie as we want as long we don't overwrite them when "switching" between the environments.

For the check "if production", I would use the constants defined in the solution center communicator.

What if ?
```
var TOKEN_COOKIE_KEY = self.getTokenCookieKey();
```

Implementing the logic at the provider level, since he knows the environment? WDYT?

**Comment by gregdeane** [comment]
@rbarilani The code example I included does make use of the communicator (e.g. `.NAME`). I'm okay with moving the logic to a separate function though.

**Comment by gregdeane** [comment]
@rbarilani feel free to submit a PR

**Comment by rbarilani** [comment]
@gregdeane Yep I will try to find a time for that.
